src/main/scala/com/campudus/tableaux/cache/CacheVerticle.scala

```diff
@@ -0,0 +1,202 @@
+package com.campudus.tableaux.cache
+
+import java.util.concurrent.TimeUnit
+
+import com.campudus.tableaux.database.model.TableauxModel.{ColumnId, TableId}
+import com.google.common.cache.CacheBuilder
+import io.vertx.core.eventbus.Message
+import io.vertx.scala.ScalaVerticle
+import org.vertx.scala.core.json.{Json, JsonObject}
+
+import scala.collection.mutable
+import scala.concurrent.{Future, Promise}
+import scala.language.implicitConversions
+import scalacache._
+import scalacache.guava._
+import scalacache.serialization.InMemoryRepr
+
+object CacheVerticle {
+  val NOT_FOUND_FAILURE = 404
+  val INVALID_MESSAGE = 400
+
+  val ADDRESS_SET = "cache.set"
+  val ADDRESS_RETRIEVE = "cache.retrieve"
+  val ADDRESS_INVALIDATE = "cache.invalidate"
+  val ADDRESS_INVALIDATE_ALL = "cache.invalidate_all"
+}
+
+class CacheVerticle extends ScalaVerticle {
+
+  import CacheVerticle._
+
+  lazy val eventBus = vertx.eventBus()
+
+  val caches: mutable.Map[(TableId, ColumnId), ScalaCache[InMemoryRepr]] = mutable.Map.empty
+
+  override def start(promise: Promise[Unit]): Unit = {
+    registerOnEventBus()
+
+    promise.success(())
+  }
+
+  override def stop(promise: Promise[Unit]): Unit = {
+    promise.success(())
+  }
+
+  private def registerOnEventBus(): Unit = {
+    import io.vertx.scala.FunctionConverters._
+
+    eventBus.localConsumer(ADDRESS_SET, messageHandlerSet(_: Message[JsonObject]))
+    eventBus.localConsumer(ADDRESS_RETRIEVE, messageHandlerRetrieve(_: Message[JsonObject]))
+    eventBus.localConsumer(ADDRESS_INVALIDATE, messageHandlerInvalidate(_: Message[JsonObject]))
+    eventBus.localConsumer(ADDRESS_INVALIDATE_ALL, messageHandlerInvalidateAll(_: Message[JsonObject]))
+  }
+
+  private def getCache(tableId: TableId, columnId: ColumnId): ScalaCache[InMemoryRepr] = {
+    def createCache() = CacheBuilder
+      .newBuilder()
+      .expireAfterAccess(config().getLong("expireAfterAccess", 3600l).toLong, TimeUnit.SECONDS)
+      .maximumSize(config().getLong("maximumSize", 10000l).toLong)
```

Comment:
Maybe put the `expireAfterAccess` and `maximumSize` settings into variables. Looks like magic numbers to me here.